components/data_comps/dlnd/src/dlnd_comp_mod.F90

```diff
@@ -256,6 +278,119 @@ subroutine dlnd_comp_init(Eclock, x2l, l2x, &
 
     call t_stopf('dlnd_initmctdom')
 
+#ifdef HAVE_MOAB
+    ilat = mct_aVect_indexRA(ggrid%data,'lat')
+    ilon = mct_aVect_indexRA(ggrid%data,'lon')
+    allocate(moab_vert_coords(lsize*3))
+    do iv = 1, lsize
+       lonv = ggrid%data%rAttr(ilon, iv) * SHR_CONST_PI/180.
+       latv = ggrid%data%rAttr(ilat, iv) * SHR_CONST_PI/180.
```

Comment:
We should expose a utility method in MOAB to do coordinate transformations. It doesn't make sense having this exact same type of transform code in every component

Reply:
We have this in MOAB, but it does not make sense to expose it from there; iMOAB?  The utility should be either in 
E3SM/share or in E3SM/driver-moab/shr

Reply:
If its only used within the coupler code (including the code in the models) then driver-moab/shr would be the place.

Reply:
But you don't have to do that for this PR.